tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -849,87 +857,31 @@ string TRT_ShapedWeights::DebugString() const {
-      : trt_dtype_(trt_dtype), trt_dims_(trt_dims) {}
-
-  void setName(const char* name) override {}
-
-  const char* getName() const override { return ""; }
-
-  void setDimensions(nvinfer1::Dims dimensions) override {
-    trt_dims_ = dimensions;
-  }
-
-  nvinfer1::Dims getDimensions() const override { return trt_dims_; }
-
-  void setType(nvinfer1::DataType trt_dtype) override {
-    trt_dtype_ = trt_dtype;
-  }
+// SimpleITensor implementation moved to Proxy ITensor class file.
+// See tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h
 
-  nvinfer1::DataType getType() const override { return trt_dtype_; }
-
-  bool isNetworkInput() const override { return false; }
-
-  bool isNetworkOutput() const override { return false; }
-
-  void setBroadcastAcrossBatch(bool broadcastAcrossBatch) override {}
-
-  bool getBroadcastAcrossBatch() const override { return false; }
-
-  nvinfer1::TensorLocation getLocation() const override {
-    // This is arbitrary, since we don't use it.
-    return nvinfer1::TensorLocation::kDEVICE;
-  }
-
-  void setLocation(nvinfer1::TensorLocation location) override {}
-
-#if IS_TRT_VERSION_GE(5, 0, 0, 0)
-  bool setDynamicRange(float min, float max) override { return true; }
-
-  float getDynamicRange() const override { return 0; }
-#endif
-
-#if IS_TRT_VERSION_GE(5, 1, 0, 0)
-  bool dynamicRangeIsSet() const override { return true; }
-
-  void resetDynamicRange() override {}
-
-  float getDynamicRangeMin() const override { return 0.f; }
-
-  float getDynamicRangeMax() const override { return 0.f; }
-#endif
-
-#if IS_TRT_VERSION_GE(6, 0, 0, 0)
-  void setAllowedFormats(nvinfer1::TensorFormats formats) override {}
-
-  nvinfer1::TensorFormats getAllowedFormats() const override { return 1; }
-
-  bool isShapeTensor() const override { return false; }
-
-  bool isExecutionTensor() const override { return true; }
-#endif
+TRT_TensorOrWeights::TRT_TensorOrWeights(ITensorProxyPtr tensor)
+    : tensor_proxy_ptr_(tensor),
+      initialized_(true),
+      is_tensor_(true) {}
 
- private:
-  nvinfer1::DataType trt_dtype_;
-  nvinfer1::Dims trt_dims_;
-};
+TRT_TensorOrWeights::TRT_TensorOrWeights(ITensorProxyPtr tensor, int batch_size)
+    : tensor_proxy_ptr_(tensor),
+      batch_size_(batch_size),
+      initialized_(true),
+      is_tensor_(true) {}
 
 TRT_TensorOrWeights::TRT_TensorOrWeights(nvinfer1::ITensor* tensor,
                                          int batch_size)
-    : tensor_(tensor),
+    : tensor_proxy_ptr_(tensor),
       batch_size_(batch_size),
       initialized_(true),
       is_tensor_(true) {}
 
 TRT_TensorOrWeights::TRT_TensorOrWeights(nvinfer1::DataType trt_dtype,
                                          const nvinfer1::Dims& trt_dims,
                                          int batch_size)
-    : simple_itensor_(new SimpleITensor(trt_dtype, trt_dims)),
+    : tensor_proxy_ptr_(new SimpleITensor(trt_dtype, trt_dims)),
```

Comment:
Who is going to release the memory?
Similar in other places where you  use "new", we need to make sure that the memory will be released.